test/optimizer/plan_selection_test.cpp

```diff
@@ -0,0 +1,607 @@
+
+    auto plan = optimizer.BuildPelotonPlanTree(stmt, DEFAULT_DB_NAME, txn);
+    timer_.Start();
+    txn_manager.CommitTransaction(txn);
+    timer_.Stop();
+    LOG_INFO("Query Execution Duration: %f", timer_.GetDuration());
+    PrintPlan(plan);
+
+    return plan;
+  }
+
+  std::string CreateTwoWayJoinQuery(const std::string &table_1,
+                                    const std::string &table_2,
+                                    const std::string &column_1,
+                                    const std::string &column_2) {
+    return CreateTwoWayJoinQuery(table_1, table_2, column_1, column_2, "", "");
+  }
+
+  std::string CreateTwoWayJoinQuery(const std::string &table_1,
+                                    const std::string &table_2,
+                                    const std::string &column_1,
+                                    const std::string &column_2,
+                                    const std::string &order_by_table,
+                                    const std::string &order_by_column) {
+    std::stringstream ss;
+    ss << "SELECT * FROM " << table_1 << ", " << table_2 << " WHERE " << table_1
+       << "." << column_1 << " = " << table_2 << "." << column_2;
+    if (!order_by_column.empty() and !order_by_table.empty()) {
+      ss << " ORDER BY " << order_by_table << "." << order_by_column;
+    }
+    ss << ";";
+    return ss.str();
+  }
+
+  void InsertDataHelper(const std::string &table_name, int tuple_count) {
+    int batch_size = 1000;
+    std::stringstream ss;
+    auto count = 0;
+    if (tuple_count > batch_size) {
+      for (int i = 0; i < tuple_count; i += batch_size) {
+        ss.str(std::string());
+        ss << "INSERT INTO " << table_name << " VALUES ";
+        for (int j = 1; j <= batch_size; j++) {
+          count++;
+          ss << "(" << count << ", 1.1, 'abcd')";
+          if (j < batch_size) {
+            ss << ",";
+          }
+        }
+        ss << ";";
+        TestingSQLUtil::ExecuteSQLQuery(ss.str());
+      }
+    } else {
+      ss << "INSERT INTO " << table_name << " VALUES ";
+      for (int i = 1; i <= tuple_count; i++) {
+        ss << "(" << i << ", 1.1, 'abcd')";
+        if (i < tuple_count) {
+          ss << ",";
+        }
+      }
+      ss << ";";
+      TestingSQLUtil::ExecuteSQLQuery(ss.str());
+    }
+    LOG_INFO("COUNT: %d", count);
+  }
+
+  void AnalyzeTable(const std::string &table_name) {
+    std::stringstream ss;
+    ss << "ANALYZE " << table_name << ";";
+    TestingSQLUtil::ExecuteSQLQuery(ss.str());
+  }
+
+  void PrintPlan(std::shared_ptr<planner::AbstractPlan> plan, int level = 0) {
+    PrintPlan(plan.get(), level);
+  }
+
+  void PrintPlan(planner::AbstractPlan *plan, int level = 0) {
+    auto spacing = std::string(level, '\t');
+
+    if (plan->GetPlanNodeType() == PlanNodeType::SEQSCAN) {
+      auto scan = dynamic_cast<planner::AbstractScan *>(plan);
+      LOG_DEBUG("%s%s(%s)", spacing.c_str(), scan->GetInfo().c_str(),
```

Comment:
Noe sure if we need to consider this, but `scan` variable will be throwing unused variable error if the build mode is set to Release.